macros/light-sdk-macros/tests/hasher.rs

```diff
@@ -15,64 +16,1202 @@ pub struct MyAccount {
+        use super::*;
+
+        #[test]
+        fn test_byte_representation() {
+            let nested_struct = create_nested_struct();
+            let account = create_account(Some(42));
+
+            let manual_nested_bytes: Vec<Vec<u8>> = vec![
+                nested_struct.a.to_le_bytes().to_vec(),
+                nested_struct.b.to_le_bytes().to_vec(),
+                light_utils::hash_to_bn254_field_size_be(nested_struct.c.as_bytes())
+                    .unwrap()
+                    .0
+                    .to_vec(),
+            ];
+
+            let nested_bytes: Vec<&[u8]> =
+                manual_nested_bytes.iter().map(|v| v.as_slice()).collect();
+            let manual_nested_hash = Poseidon::hashv(&nested_bytes).unwrap();
+
+            let nested_reference_hash = [
+                6, 124, 124, 67, 65, 28, 217, 111, 86, 61, 85, 93, 118, 177, 69, 25, 117, 70, 49,
+                96, 28, 232, 61, 133, 166, 55, 135, 210, 49, 27, 114, 93,
+            ];
+            let nested_hash_result = nested_struct.hash::<Poseidon>().unwrap();
+
+            assert_eq!(nested_struct.as_byte_vec(), manual_nested_bytes);
+            assert_eq!(nested_hash_result, nested_reference_hash);
+            assert_eq!(manual_nested_hash, nested_reference_hash);
+            assert_eq!(nested_hash_result, manual_nested_hash);
+
+            let manual_account_bytes: Vec<Vec<u8>> = vec![
+                vec![u8::from(account.a)],
+                account.b.to_le_bytes().to_vec(),
+                account.c.hash::<Poseidon>().unwrap().to_vec(),
+                light_utils::hash_to_bn254_field_size_be(&account.d)
+                    .unwrap()
+                    .0
+                    .to_vec(),
+                {
+                    let mut bytes = vec![1u8]; // Prefix with 1 for Some
+                    bytes.extend_from_slice(&account.f.unwrap().to_le_bytes());
+                    bytes
+                },
+            ];
+
+            let account_bytes: Vec<&[u8]> =
+                manual_account_bytes.iter().map(|v| v.as_slice()).collect();
+            let manual_account_hash = Poseidon::hashv(&account_bytes).unwrap();
+
+            let account_hash_result = account.hash::<Poseidon>().unwrap();
+
+            assert_eq!(account.as_byte_vec(), manual_account_bytes);
+            assert_eq!(account_hash_result, manual_account_hash);
+        }
+
+        #[test]
+        fn test_boundary_values() {
+            let nested = create_zero_nested();
+
+            let zero_account = MyAccount {
+                a: false,
+                b: 0,
+                c: nested,
+                d: [0; 32],
+                e: MyNestedNonHashableStruct {
+                    a: PhantomData,
+                    b: Rc::new(RefCell::new(0)),
+                },
+                f: Some(0),
+            };
+
+            let hash = zero_account.hash::<Poseidon>().unwrap();
+            assert_eq!(hash.len(), 32);
```

Comment:
this assert doesn't make sense to me since there is no case where the hash can be of a different length than 32 bytes.
```suggestion
              let manual_account_bytes: Vec<Vec<u8>> = vec![
                  vec![u8::from(zero_account.a)],
                  zero_account.b.to_le_bytes().to_vec(),
                  zero_account.c.hash::<Poseidon>().unwrap().to_vec(),
                  light_utils::hash_to_bn254_field_size_be(&zero_account.d)
                      .unwrap()
                      .0
                      .to_vec(),
                  zero_account.e.b.to_le_bytes().to_vec(),
                  {
                      let mut bytes = vec![1u8];
                      bytes.extend_from_slice(&zero_account.f.unwrap().to_le_bytes());
                      bytes
                  },
              ];
            let account_bytes: Vec<&[u8]> =
                manual_account_bytes.iter().map(|v| v.as_slice()).collect();
            let manual_account_hash = Poseidon::hashv(&account_bytes).unwrap();
            let hash = zero_account.hash::<Poseidon>().unwrap();
            assert_eq!(hash, manual_account_hash);
```